### DISCUSSION TITLE
Get correct permissions in hook for usersService

### DISCUSSION THREAD

**Discussion by sapientorius** [directus/directus#18040] [Archive]
I guess this is an easy one, but I can't find the solution.

I have a hook, when a user creates an item in the collection "domain".

Then I want to get the data of the user by using the usersService, but I get a Forbidden exception. How can I handle requests in extension with all permissions?

```js
action("domains.items.create", async (metaObject, context) => {
  const { ItemsService, UsersService, RolesService } = services;

  const usersService = new UsersService({ schema: context.schema });

  //get publisher id of user which created the domain
  try {
    let result = await usersService.readOne(context.accountability.id, {
      fields: ["rep_id"],
    });
    console.log("result", result);
  } catch (err) {
    console.error(err);
    throw err;
  }
});

```

I read about setting accountability to null, but that didn't work. (const usersService = new UsersService({ schema: context.schema, accountability: null}))

**Comment by rijkvanzanten** [comment]
I think the problem here is that `context.accountability.id` doesn't exist. Can you try `context.accountability.user`? 🙂

**Reply by sapientorius** [reply]
OMG! Can you please delete this embarrassing post? :D

Thank you!
